File: composeApp/src/commonMain/kotlin/org/sesacteamproject/passmate/navigation/Route.kt
package org.sesacteamproject.passmate.navigation

// 공통 라우트 규격 (규칙 §2-1-1) — 이름·인자는 3플랫폼 동일 유지. iosApp Route.swift와 1:1
sealed class Route(val route: String) {

    data object Home : Route("home")

    // 공개 방 목록·탐색 (M-11) — 홈에서 진입하는 방 찾기 화면
    data object RoomList : Route("roomList")

    data object SignIn : Route("signIn")

    data object Join : Route("join?pin={pin}")

    data object Waiting : Route("waiting/{pin}")

    data object Play : Route("play/{pin}")

    // 결과·리포트 화면 — 백엔드 /rooms/{roomId}/results/me 기준으로 roomId를 인자로 받는다
    // (2026-08-28 백엔드 명세서 정합. 마이페이지(파트2 US6)도 roomId로 진입한다)
    data object Result : Route("result/{roomId}")

    data object MyInfo : Route("myInfo")

    // 참여한 방 탭 루트 (M-08) — 하단 4탭 셸
    data object JoinedRooms : Route("joinedRooms")

    // 내 명성·뱃지 상세 (M-09) — 마이 탭 프로필 카드에서 진입
    data object Reputation : Route("reputation")

    // 내가 만든 방 (M-13) — 하단 탭 루트, 새 방 만들기 시트 포함
    data object HostedRooms : Route("hostedRooms")

    // 방 리포트 (M-14) — 내가 만든 방 › 종료 › 상세
    data object RoomReport : Route("roomReport/{roomId}")

    // 진행 리모컨 (M-T2) — 내가 만든 방 › 진행 중 › 진행
    data object SessionControl : Route("sessionControl/{roomId}/{pin}")

    // 정산 (M-T4) — 마이 탭에서 진입, 정산 계좌 시트(M-12-3) 포함
    data object Earnings : Route("earnings")

    data object Payment : Route("payment/{pin}")

    // 코인·결제 내역 (M-12) — 마이 탭에서 진입
    data object CoinHistory : Route("coinHistory")

    // 코인 충전 (M-12-4·M-12-6) — 마이 탭 보유 코인 행에서 진입
    data object CoinCharge : Route("coinCharge")

    data object Settings : Route("settings")
}

File: iosApp/iosApp/navigation/Route.swift
// 공통 라우트 규격 (규칙 §2-1-1) — 이름·인자는 3플랫폼 동일 유지. Compose Route.kt와 1:1
enum Route: Hashable {
    case home
    case roomList
    case signIn
    case join(pin: String?)
    case waiting(pin: String)
    case play(pin: String)
    case result(roomId: Int64)
    case myInfo
    case joinedRooms
    case reputation
    case hostedRooms
    case roomReport(roomId: Int64)
    case sessionControl(roomId: Int64, pin: String)
    case earnings
    case payment(pin: String)
    case coinHistory
    case coinCharge
    // 회원 탈퇴 (M-12-12) — 마이 › 회원 탈퇴 행에서 진입
    case deleteAccount
    // 계정 정보 변경 (M-12-1) — 닉네임·이메일 + 캐릭터 바꾸기 링크
    case editProfile
    // 내 캐릭터 변경 (M-12-7) — M-12-1의 "캐릭터 바꾸기 →"에서 진입
    case characterEdit
    // 정산 계좌 등록 (M-12-3) — 마이 · 정산(M-T4) 양쪽에서 진입
    case settlementAccount
    // 결제 수단 관리 (M-12-8)
    case paymentMethod
    // 알림 설정 (M-12-10)
    case notificationSettings

    // 탭 루트에서 push되지만 시안이 하단 탭바를 유지하는 화면 (M-12-x 전부 · M-14 방 리포트).
    // Compose AppTab.barOwnerOf와 같은 규칙이다 — nil이면 탭바를 숨긴다
    var tabBarOwner: AppTab? {
        switch self {
        case .editProfile, .characterEdit, .settlementAccount, .coinCharge,
             .paymentMethod, .coinHistory, .notificationSettings, .deleteAccount:
            return .myInfo
        case .roomReport:
            return .hostedRooms
        default:
            return nil
        }
    }

    // 세션 플로우 엔트리(Join·Payment·Waiting·Play) — Result 진입 시 이것만 제거하고 탭 루트는 유지한다 (규칙 §2-1-2, 스펙 §1-5)
    var isSessionRoute: Bool {
        switch self {
        case .join, .payment, .waiting, .play: return true
        default: return false
        }
    }
}
